lib/src/reduction_tree.dart

```diff
@@ -83,41 +93,43 @@ class ReductionTree extends Module {
       Logic? enable,
       Logic? reset,
       super.name = 'reduction_tree'})
-      : super(definitionName: 'ReductionTree_R${radix}_L${sequence.length}}') {
-    if (sequence.isEmpty) {
-      throw RohdHclException("Don't use ReductionTree "
-          'with an empty sequence');
-    }
-    sequence = [
+      : super(
+            definitionName: 'ReductionTreeNode_R${radix}_L${sequence.length}') {
+    this.sequence = [
       for (var i = 0; i < sequence.length; i++)
         addInput('seq$i', sequence[i], width: sequence[i].width)
     ];
     this.clk = (clk != null) ? addInput('clk', clk) : null;
     this.enable = (enable != null) ? addInput('enable', enable) : null;
     this.reset = (reset != null) ? addInput('reset', reset) : null;
 
-    _computed = reductionTreeRecurse(sequence);
-    addOutput('out', width: _computed.value.width) <= _computed.value;
-  }
-
-  /// Local conditional flop using module reset/enable
-  Logic localFlop(Logic d, {bool doFlop = false}) =>
-      condFlop(doFlop ? clk : null, reset: reset, en: enable, d);
-
-  /// Recursively construct the computation tree
-  ({Logic value, int depth, int flopDepth}) reductionTreeRecurse(
-      List<Logic> seq) {
-    if (seq.length < radix) {
-      return (value: operation(seq), depth: 0, flopDepth: 0);
+    if (this.sequence.length <= radix) {
+      final value = operation(this.sequence);
+      addOutput('out', width: value.width) <= value;
+      _computed = (value: output('out'), depth: 0, flopDepth: 0);
     } else {
       final results = <({Logic value, int depth, int flopDepth})>[];
-      final segment = seq.length ~/ radix;
+      final segment = this.sequence.length ~/ radix;
+
+      // final divisor = (log(sequence.length - 1) / log(radix)).floor();
+      // final segment = pow(radix, divisor).toInt();
+
       var pos = 0;
       for (var i = 0; i < radix; i++) {
-        final c = reductionTreeRecurse(seq
-            .getRange(pos, (i < radix - 1) ? pos + segment : seq.length)
-            .toList());
-        results.add(c);
+        final tree = ReductionTree(
+            this
+                .sequence
+                .getRange(
+                    pos, (i < radix - 1) ? pos + segment : this.sequence.length)
+                .toList(),
+            operation,
+            radix: radix,
+            signExtend: signExtend,
+            depthToFlop: depthToFlop,
+            clk: this.clk,
```

Comment:
suggestion: keeping this logic generation in another function (e.g. `_buildLogic()`) makes it easier to avoid accidentally using the wrong version of variables without needing to put `this`.  Additionally, you could consider making some of these variable (e.g. clk, enable, reset, sequence) private instead of protected, unless there's a reason to extend this.